File: src/model/characters/CharacterImpl.java
package src.model.characters;

public class CharacterImpl implements Character{

    private String name;
    private int belongsTo;
    public int health;
    public boolean ate;

    CharacterImpl(String name, int ID) {
        this.name = name;
        this.belongsTo = ID;
    }

    public String getName() {
        return this.name;
    }

    public int owner() {
        return this.belongsTo;
    }
}

File: src/newModel/characters/CharacterImpl.ts
import { Character } from "../../types/tcgTypes";

export class CharacterImpl implements Character {
    private _name:String;
    private _belongsTo:number;
    public health:number;
    public ate:boolean;

    constructor(name:String, ID:number) {
        this._name = name;
        this._belongsTo = ID;
    }
    
    get name(): String {
        return this._name;
    }

    get owner(): number {
        return this._belongsTo;
    }
}